proto/src/serialize/binary/encoder.rs

```diff
@@ -15,12 +15,13 @@
  */
 use std::collections::HashMap;
 use std::sync::Arc as Rc;
+use byteorder::{ByteOrder, NetworkEndian};
 
 use error::{ProtoErrorKind, ProtoResult};
 
 /// Encode DNS messages and resource record types.
 pub struct BinEncoder<'a> {
-    offset: u32,
+    offset: usize,
```

Comment:
not sure if this is really helpful, but otherwise we have to cast in `write_slice`. The cost is probably negligeable though.

Reply:
I'm ok with which ever ends up with fewer casts...

Reply:
we definitely have fewer casts with `usize` then, since `write_slice` is called very often.